### PR TITLE
Restore nodes from individual EBS snapshots

### PR DESCRIPTION
**Pull request by flfernandez**
This PR is to add the capability of creating a new cluster with the secondary EBS (auto_volumes) created from separate snapshots, i.e. not a single EBS snapshot as source for the volumes of a cluster.

The snapshot is filter by Tags, example: tag:backup_id

The use case is for a cluster with the same amount of nodes (same topology) as the original one. For example the case of a 3 node cluster, filtering the snapshots by Tags will return 3 snapshots IDs to be restored into the corresponding 3 nodes.

### CODE REVIEW

File: _dependencies/tasks/main.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@
...
+#     - assert: { that: "app_class is defined and app_class != ''", msg: "Please define app_class" }
+#     - assert: { that: "clusterid is defined and cluster_vars is defined", msg: "Please define clusterid" }
+#     - assert: { that: "buildenv is defined and cluster_vars[buildenv] is defined", msg: "Please define buildenv" }
 
-    - assert: { that: "(cluster_vars.assign_public_ip == 'yes' and cluster_vars.inventory_ip == 'public') or (cluster_vars.inventory_ip == 'private')", msg: "If inventory_ip=='public', 'assign_public_ip' must be 'yes'" }
-      when: cluster_vars.type == "gce" or cluster_vars.type == "aws"
-  tags: clusterverse_clean,clusterverse_create,clusterverse_dynamic_inventory,clusterverse_config,clusterverse_readiness
+#     - assert: { that: "(cluster_vars.assign_public_ip == 'yes' and cluster_vars.inventory_ip == 'public') or (cluster_vars.inventory_ip == 'private')", msg: "If inventory_ip=='public', 'assign_public_ip' must be 'yes'" }
+#       when: cluster_vars.type == "gce" or cluster_vars.type == "aws"
+#   tags: clusterverse_clean,clusterverse_create,clusterverse_dynamic_inventory,clusterverse_config,clusterverse_readiness
```
Comment (leslieonline27) [inline _dependencies/tasks/main.yml]: Why are we commenting out here?

File: cluster_hosts/tasks/main.yml [hunk 2]
```diff
@@ -4,29 +4,37 @@
   set_fact:
     epoch_time: "{{ansible_date_time.epoch}}"
 
-# Create an array of dictionaries containing all the hostnames PER-AZ (i.e. couchbase-dev-node-a0, couchbase-dev-master-a1, couchbase-dev-master-b0, couchbase-dev-master-b1 etc) to be created:
```
Comment (mikekrasn) [inline cluster_hosts/tasks/main.yml]: has to be reverted

File: _dependencies/tasks/main.yml (hunk 1 above)
Comment (mikekrasn) [reply _dependencies/tasks/main.yml]: That's my bed, forgotten to revert that.

File: cluster_hosts/tasks/main.yml [hunk 3]
```diff
@@ -4,29 +4,37 @@
...
 - include_tasks: aws.yml
   when: cluster_vars.type == "aws"
 
 - include_tasks: gce.yml
   when: cluster_vars.type == "gce"
 
 #- debug: msg={{cluster_hosts_flat}}
+
+
+# Correcting @eyettea 's clean solution: Another solution that's a bit cleaner imo is to initialize an empty list {% set count = [ ] %}, add an item to the list in every loop {% set __ = count.append(1) %} and use the length to display the count count|length
```
Comment (mikekrasn) [inline cluster_hosts/tasks/main.yml]: has to be removed

**Pull request merged**